packages/editor/src/store/private-selectors.js

```diff
@@ -203,3 +176,61 @@ export const getPostBlocksByName = createRegistrySelector( ( select ) =>
 		() => [ select( blockEditorStore ).getBlocks() ]
 	)
 );
+
+export const getPostTitle = createRegistrySelector(
+	( select ) => ( state, postType, postId ) => {
+		const _document = select( coreStore ).getEditedEntityRecord(
+			'postType',
+			postType,
+			postId
+		);
+		const isTemplate = TEMPLATE_POST_TYPES.includes( postType );
+		const { title: defaultTitle } =
+			isTemplate && _document?.slug
+				? __experimentalGetDefaultTemplateType( state, _document?.slug )
+				: {};
+		const documentTitle =
+			typeof _document?.title === 'string'
+				? _document.title
+				: title?.rendered;
+		const title =
+			isTemplate &&
+			( ! documentTitle || documentTitle === _document?.slug )
+				? defaultTitle
+				: documentTitle;
+
+		return title ? decodeEntities( title ) : __( 'No title' );
```

Comment:
Maybe we should leave decoding and providing a fallback to consumers.

Reply:
Why? I hesitated personally but after thinking about it, I don't know why would anyone want an encoded value. More arguable for the fallback value but this allows to have a consistent fallback value.

Reply:
> I hesitated personally but after thinking about it, I don't know why would anyone want an encoded value.
I have no idea, to be honest 😄  But my take on entity decoding is similar to data escaping. Do it when you echo, not before.

> More arguable for the fallback value but this allows to have a consistent fallback value.
What if the component wants a different fallback? They'll have to duplicate logic again.

These are not blockers, just for general consideration.